```
Error running mint automation
+ mkdir -p 8550-d0e03c5/gopath/src/github.com/minio
+ git -C 8550-d0e03c5/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 8550-d0e03c5/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 8550-d0e03c5/gopath/src/github.com/minio/minio fetch --quiet minio pull/8550/head:pr8550
+ git -C 8550-d0e03c5/gopath/src/github.com/minio/minio checkout --quiet pr8550
+ GO111MODULE=on
+ GOPATH=/mnt/nvme/mint-auto/8550-d0e03c5/gopath
+ make -C 8550-d0e03c5/gopath/src/github.com/minio/minio --quiet
go: downloading github.com/minio/gokrb5/v7 v7.2.5
go: downloading github.com/minio/hdfs/v3 v3.0.1
go: downloading github.com/kurin/blazer v0.5.4-0.20190613185654-cf2f27cc0be3
go: downloading github.com/minio/mc v0.0.0-20191012041914-735aa139b19c
go: downloading github.com/minio/minio-go v0.0.0-20190327203652-5325257a208f
go: downloading github.com/gorilla/mux v1.7.0
go: downloading github.com/aliyun/aliyun-oss-go-sdk v0.0.0-20190307165228-86c17b95fcd5
go: downloading github.com/json-iterator/go v1.1.7
go: downloading github.com/pkg/profile v1.3.0
go: extracting github.com/gorilla/mux v1.7.0
go: downloading github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
go: extracting github.com/pkg/profile v1.3.0
go: downloading github.com/minio/sio v0.2.0
go: extracting github.com/minio/minio-go v0.0.0-20190327203652-5325257a208f
go: downloading github.com/minio/minio-go/v6 v6.0.39
go: extracting github.com/aliyun/aliyun-oss-go-sdk v0.0.0-20190307165228-86c17b95fcd5
go: extracting github.com/minio/sio v0.2.0
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: extracting github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
go: extracting github.com/minio/mc v0.0.0-20191012041914-735aa139b19c
go: downloading google.golang.org/api v0.4.0
go: extracting github.com/json-iterator/go v1.1.7
go: extracting github.com/minio/minio-go/v6 v6.0.39
go: extracting github.com/kurin/blazer v0.5.4-0.20190613185654-cf2f27cc0be3
go: downloading github.com/minio/sha256-simd v0.1.1
go: downloading github.com/fatih/color v1.7.0
go: downloading github.com/skyrings/skyring-common v0.0.0-20160929130248-d1c0bb1cbd5e
go: downloading github.com/prometheus/common v0.2.0
go: extracting github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading github.com/coreos/etcd v3.3.12+incompatible
go: extracting github.com/fatih/color v1.7.0
go: extracting github.com/minio/sha256-simd v0.1.1
go: downloading github.com/mattn/go-colorable v0.1.1
go: downloading github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: extracting github.com/minio/hdfs/v3 v3.0.1
go: extracting github.com/prometheus/common v0.2.0
go: downloading github.com/minio/lsync v1.0.1
go: downloading golang.org/x/net v0.0.0-20190923162816-aa69164e4478
go: extracting github.com/minio/lsync v1.0.1
go: downloading go.uber.org/atomic v1.3.2
go: extracting github.com/mattn/go-colorable v0.1.1
go: downloading github.com/klauspost/compress v1.8.3
go: extracting github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: downloading github.com/gorilla/rpc v1.2.0
go: extracting github.com/gorilla/rpc v1.2.0
go: extracting go.uber.org/atomic v1.3.2
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: extracting golang.org/x/net v0.0.0-20190923162816-aa69164e4478
go: downloading github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
go: downloading github.com/klauspost/reedsolomon v1.9.3
go: extracting github.com/matttproud/golang_protobuf_extensions v1.0.1
go: downloading github.com/shirou/gopsutil v2.18.12+incompatible
go: extracting github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
go: downloading github.com/minio/highwayhash v1.0.0
go: extracting github.com/klauspost/reedsolomon v1.9.3
go: downloading golang.org/x/sys v0.0.0-20190922100055-0a153f010e69
go: extracting github.com/coreos/etcd v3.3.12+incompatible
go: extracting github.com/shirou/gopsutil v2.18.12+incompatible
go: downloading golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: extracting github.com/minio/highwayhash v1.0.0
go: extracting golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: downloading github.com/coredns/coredns v1.4.0
go: downloading gopkg.in/ini.v1 v1.48.0
go: extracting github.com/skyrings/skyring-common v0.0.0-20160929130248-d1c0bb1cbd5e
go: downloading github.com/mattn/go-isatty v0.0.7
go: extracting gopkg.in/ini.v1 v1.48.0
go: downloading github.com/prometheus/procfs v0.0.0-20190117184657-bf6a532e95b1
go: extracting github.com/coredns/coredns v1.4.0
go: downloading github.com/cheggaaa/pb v1.0.28
go: extracting github.com/mattn/go-isatty v0.0.7
go: downloading github.com/klauspost/readahead v1.3.1
go: extracting github.com/cheggaaa/pb v1.0.28
go: extracting github.com/klauspost/readahead v1.3.1
go: downloading golang.org/x/crypto v0.0.0-20190923035154-9ee001bba392
go: extracting github.com/prometheus/procfs v0.0.0-20190117184657-bf6a532e95b1
go: downloading github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: extracting golang.org/x/sys v0.0.0-20190922100055-0a153f010e69
go: downloading github.com/miekg/dns v1.1.8
go: extracting github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: downloading github.com/gomodule/redigo v2.0.0+incompatible
go: extracting github.com/miekg/dns v1.1.8
go: extracting golang.org/x/crypto v0.0.0-20190923035154-9ee001bba392
go: extracting github.com/gomodule/redigo v2.0.0+incompatible
go: downloading github.com/elazarl/go-bindata-assetfs v1.0.0
go: extracting github.com/elazarl/go-bindata-assetfs v1.0.0
go: downloading github.com/klauspost/cpuid v1.2.1
go: downloading github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: extracting github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: downloading github.com/Azure/go-autorest v11.7.0+incompatible
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: extracting github.com/klauspost/cpuid v1.2.1
go: downloading github.com/minio/parquet-go v0.0.0-20190318185229-9d767baf1679
go: extracting github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading gopkg.in/Shopify/sarama.v1 v1.20.0
go: extracting github.com/Azure/go-autorest v11.7.0+incompatible
go: downloading gopkg.in/yaml.v2 v2.2.2
go: extracting github.com/minio/parquet-go v0.0.0-20190318185229-9d767baf1679
go: downloading github.com/fatih/structs v1.1.0
go: extracting gopkg.in/yaml.v2 v2.2.2
go: extracting gopkg.in/Shopify/sarama.v1 v1.20.0
go: extracting github.com/fatih/structs v1.1.0
go: downloading github.com/minio/cli v1.22.0
go: downloading github.com/pkg/errors v0.8.1
go: extracting github.com/pkg/errors v0.8.1
go: downloading github.com/dustin/go-humanize v1.0.0
go: extracting github.com/minio/cli v1.22.0
go: extracting github.com/dustin/go-humanize v1.0.0
go: downloading github.com/streadway/amqp v0.0.0-20190402114354-16ed540749f6
go: extracting github.com/streadway/amqp v0.0.0-20190402114354-16ed540749f6
go: downloading github.com/Azure/azure-sdk-for-go v33.4.0+incompatible
go: extracting github.com/klauspost/compress v1.8.3
go: downloading google.golang.org/grpc v1.20.1
go: extracting google.golang.org/api v0.4.0
go: extracting google.golang.org/grpc v1.20.1
go: downloading github.com/nsqio/go-nsq v1.0.7
go: downloading google.golang.org/genproto v0.0.0-20190307195333-5fe7a883aa19
go: extracting github.com/nsqio/go-nsq v1.0.7
go: downloading git.apache.org/thrift.git v0.12.0
go: downloading github.com/eapache/queue v1.1.0
go: downloading github.com/gogo/protobuf v1.2.1
go: extracting github.com/eapache/queue v1.1.0
go: downloading github.com/secure-io/sio-go v0.3.0
go: extracting github.com/secure-io/sio-go v0.3.0
go: downloading github.com/mattn/go-runewidth v0.0.4
go: extracting github.com/mattn/go-runewidth v0.0.4
go: extracting github.com/gogo/protobuf v1.2.1
go: downloading golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: extracting golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: downloading github.com/golang/protobuf v1.3.2
go: extracting git.apache.org/thrift.git v0.12.0
go: extracting github.com/golang/protobuf v1.3.2
go: downloading github.com/davecgh/go-spew v1.1.1
go: extracting github.com/davecgh/go-spew v1.1.1
go: downloading github.com/djherbis/atime v1.0.0
go: extracting github.com/djherbis/atime v1.0.0
go: downloading github.com/rs/cors v1.6.0
go: extracting github.com/rs/cors v1.6.0
go: downloading github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: extracting github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: extracting github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/eclipse/paho.mqtt.golang v1.2.0
go: extracting github.com/eclipse/paho.mqtt.golang v1.2.0
go: downloading github.com/ncw/directio v1.0.5
go: extracting github.com/ncw/directio v1.0.5
go: downloading github.com/alecthomas/participle v0.2.1
go: extracting github.com/alecthomas/participle v0.2.1
go: downloading golang.org/x/text v0.3.2
go: extracting google.golang.org/genproto v0.0.0-20190307195333-5fe7a883aa19
go: downloading github.com/lib/pq v1.0.0
go: extracting github.com/lib/pq v1.0.0
go: downloading github.com/eapache/go-resiliency v1.2.0
go: extracting github.com/eapache/go-resiliency v1.2.0
go: downloading github.com/klauspost/pgzip v1.2.1
go: extracting github.com/klauspost/pgzip v1.2.1
go: downloading github.com/go-sql-driver/mysql v1.4.1
go: extracting github.com/go-sql-driver/mysql v1.4.1
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: extracting github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading github.com/nats-io/stan.go v0.4.5
go: extracting github.com/nats-io/stan.go v0.4.5
go: downloading cloud.google.com/go v0.37.2
go: extracting golang.org/x/text v0.3.2
go: downloading github.com/beevik/ntp v0.2.0
go: extracting github.com/beevik/ntp v0.2.0
go: downloading github.com/modern-go/reflect2 v1.0.1
go: extracting github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/rjeczalik/notify v0.9.2
go: extracting github.com/rjeczalik/notify v0.9.2
go: downloading go.opencensus.io v0.21.0
go: extracting cloud.google.com/go v0.37.2
go: extracting go.opencensus.io v0.21.0
go: downloading github.com/gorilla/handlers v1.4.0
go: extracting github.com/gorilla/handlers v1.4.0
go: downloading github.com/nats-io/nuid v1.0.1
go: downloading contrib.go.opencensus.io/exporter/ocagent v0.5.0
go: extracting github.com/nats-io/nuid v1.0.1
go: downloading github.com/nats-io/nats.go v1.9.1
go: extracting contrib.go.opencensus.io/exporter/ocagent v0.5.0
go: downloading github.com/pierrec/lz4 v2.0.5+incompatible
go: extracting github.com/nats-io/nats.go v1.9.1
go: downloading github.com/hashicorp/vault v1.1.0
go: extracting github.com/pierrec/lz4 v2.0.5+incompatible
go: downloading github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: extracting github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: downloading github.com/census-instrumentation/opencensus-proto v0.2.0
go: extracting github.com/census-instrumentation/opencensus-proto v0.2.0
go: downloading github.com/golang/snappy v0.0.1
go: extracting github.com/golang/snappy v0.0.1
go: downloading github.com/nats-io/nkeys v0.1.3
go: extracting github.com/nats-io/nkeys v0.1.3
go: downloading github.com/hashicorp/golang-lru v0.5.1
go: extracting github.com/hashicorp/golang-lru v0.5.1
go: downloading github.com/grpc-ecosystem/grpc-gateway v1.9.0
go: extracting github.com/grpc-ecosystem/grpc-gateway v1.9.0
go: downloading gopkg.in/olivere/elastic.v5 v5.0.80
go: extracting gopkg.in/olivere/elastic.v5 v5.0.80
go: downloading github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: extracting github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: downloading github.com/nats-io/jwt v0.3.2
go: extracting github.com/nats-io/jwt v0.3.2
go: downloading gopkg.in/ldap.v3 v3.0.3
go: extracting gopkg.in/ldap.v3 v3.0.3
go: downloading github.com/googleapis/gax-go v2.0.0+incompatible
go: extracting github.com/googleapis/gax-go v2.0.0+incompatible
go: downloading github.com/googleapis/gax-go/v2 v2.0.4
go: extracting github.com/googleapis/gax-go/v2 v2.0.4
go: downloading golang.org/x/sync v0.0.0-20190423024810-112230192c58
go: extracting golang.org/x/sync v0.0.0-20190423024810-112230192c58
go: downloading gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: extracting gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: extracting github.com/Azure/azure-sdk-for-go v33.4.0+incompatible
go: downloading github.com/satori/go.uuid v1.2.0
go: extracting github.com/satori/go.uuid v1.2.0
go: extracting github.com/hashicorp/vault v1.1.0
go: downloading github.com/hashicorp/errwrap v1.0.0
go: downloading github.com/hashicorp/go-rootcerts v1.0.1
go: downloading github.com/mitchellh/mapstructure v1.1.2
go: extracting github.com/hashicorp/errwrap v1.0.0
go: downloading github.com/hashicorp/go-multierror v1.0.0
go: extracting github.com/hashicorp/go-rootcerts v1.0.1
go: downloading github.com/hashicorp/go-retryablehttp v0.5.3
go: extracting github.com/hashicorp/go-multierror v1.0.0
go: extracting github.com/mitchellh/mapstructure v1.1.2
go: downloading github.com/hashicorp/hcl v1.0.0
go: downloading github.com/hashicorp/go-sockaddr v1.0.2
go: extracting github.com/hashicorp/go-retryablehttp v0.5.3
go: extracting github.com/hashicorp/hcl v1.0.0
go: downloading github.com/ryanuber/go-glob v1.0.0
go: downloading github.com/hashicorp/go-cleanhttp v0.5.1
go: extracting github.com/hashicorp/go-sockaddr v1.0.2
go: extracting github.com/ryanuber/go-glob v1.0.0
go: extracting github.com/hashicorp/go-cleanhttp v0.5.1
go: extracting github.com/minio/gokrb5/v7 v7.2.5
go: downloading github.com/hashicorp/go-uuid v1.0.1
go: downloading gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: downloading github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: downloading gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: downloading gopkg.in/jcmturner/aescts.v1 v1.0.1
go: downloading gopkg.in/jcmturner/rpc.v1 v1.1.0
go: extracting gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: extracting github.com/hashicorp/go-uuid v1.0.1
go: extracting gopkg.in/jcmturner/aescts.v1 v1.0.1
go: extracting gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: extracting github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: extracting gopkg.in/jcmturner/rpc.v1 v1.1.0
go: finding github.com/minio/minio-go/v6 v6.0.39
go: finding github.com/minio/gokrb5/v7 v7.2.5
go: finding google.golang.org/api v0.4.0
go: finding github.com/minio/cli v1.22.0
go: finding github.com/minio/hdfs/v3 v3.0.1
go: finding github.com/minio/sha256-simd v0.1.1
go: finding github.com/dustin/go-humanize v1.0.0
go: finding github.com/kurin/blazer v0.5.4-0.20190613185654-cf2f27cc0be3
go: finding github.com/Azure/go-autorest v11.7.0+incompatible
go: finding cloud.google.com/go v0.37.2
go: finding github.com/Azure/azure-sdk-for-go v33.4.0+incompatible
go: finding github.com/aliyun/aliyun-oss-go-sdk v0.0.0-20190307165228-86c17b95fcd5
go: finding golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: finding github.com/hashicorp/go-uuid v1.0.1
go: finding github.com/minio/mc v0.0.0-20191012041914-735aa139b19c
go: finding github.com/minio/highwayhash v1.0.0
go: finding github.com/json-iterator/go v1.1.7
go: finding github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: finding gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: finding github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: finding google.golang.org/grpc v1.20.1
go: finding github.com/gorilla/mux v1.7.0
go: finding github.com/fatih/color v1.7.0
go: finding github.com/googleapis/gax-go/v2 v2.0.4
go: finding github.com/mattn/go-colorable v0.1.1
go: finding github.com/mattn/go-isatty v0.0.7
go: finding google.golang.org/genproto v0.0.0-20190307195333-5fe7a883aa19
go: finding github.com/golang/protobuf v1.3.2
go: finding golang.org/x/sys v0.0.0-20190922100055-0a153f010e69
go: finding golang.org/x/crypto v0.0.0-20190923035154-9ee001bba392
go: finding github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: finding github.com/mitchellh/go-homedir v1.1.0
go: finding github.com/gogo/protobuf v1.2.1
go: finding golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: finding github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: finding go.opencensus.io v0.21.0
go: finding gopkg.in/ini.v1 v1.48.0
go: finding github.com/minio/minio-go v0.0.0-20190327203652-5325257a208f
go: finding contrib.go.opencensus.io/exporter/ocagent v0.5.0
go: finding github.com/satori/go.uuid v1.2.0
go: finding github.com/rjeczalik/notify v0.9.2
go: finding golang.org/x/net v0.0.0-20190923162816-aa69164e4478
go: finding github.com/modern-go/reflect2 v1.0.1
go: finding github.com/census-instrumentation/opencensus-proto v0.2.0
go: finding github.com/hashicorp/golang-lru v0.5.1
go: finding golang.org/x/text v0.3.2
go: finding gopkg.in/jcmturner/aescts.v1 v1.0.1
go: finding gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: finding github.com/grpc-ecosystem/grpc-gateway v1.9.0
go: finding github.com/shirou/gopsutil v2.18.12+incompatible
go: finding github.com/secure-io/sio-go v0.3.0
go: finding golang.org/x/sync v0.0.0-20190423024810-112230192c58
go: finding gopkg.in/jcmturner/rpc.v1 v1.1.0
go: finding github.com/ncw/directio v1.0.5
go: finding github.com/beevik/ntp v0.2.0
go: finding github.com/cheggaaa/pb v1.0.28
go: finding github.com/mattn/go-runewidth v0.0.4
go: finding github.com/coreos/etcd v3.3.12+incompatible
go: finding github.com/djherbis/atime v1.0.0
go: finding github.com/elazarl/go-bindata-assetfs v1.0.0
go: finding github.com/gorilla/handlers v1.4.0
go: finding github.com/gorilla/rpc v1.2.0
go: finding github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: finding github.com/klauspost/compress v1.8.3
go: finding github.com/klauspost/cpuid v1.2.1
go: finding github.com/klauspost/readahead v1.3.1
go: finding github.com/klauspost/reedsolomon v1.9.3
go: finding github.com/miekg/dns v1.1.8
go: finding github.com/minio/lsync v1.0.1
go: finding gopkg.in/ldap.v3 v3.0.3
go: finding gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: finding github.com/eclipse/paho.mqtt.golang v1.2.0
go: finding github.com/go-sql-driver/mysql v1.4.1
go: finding github.com/gomodule/redigo v2.0.0+incompatible
go: finding github.com/lib/pq v1.0.0
go: finding github.com/nats-io/nats.go v1.9.1
go: finding github.com/nats-io/jwt v0.3.2
go: finding github.com/nats-io/nkeys v0.1.3
go: finding github.com/nats-io/nuid v1.0.1
go: finding github.com/nats-io/stan.go v0.4.5
go: finding github.com/nsqio/go-nsq v1.0.7
go: finding github.com/golang/snappy v0.0.1
go: finding github.com/pkg/errors v0.8.1
go: finding github.com/skyrings/skyring-common v0.0.0-20160929130248-d1c0bb1cbd5e
go: finding github.com/streadway/amqp v0.0.0-20190402114354-16ed540749f6
go: finding gopkg.in/Shopify/sarama.v1 v1.20.0
go: finding github.com/davecgh/go-spew v1.1.1
go: finding github.com/eapache/go-resiliency v1.2.0
go: finding github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: finding github.com/eapache/queue v1.1.0
go: finding github.com/pierrec/lz4 v2.0.5+incompatible
go: finding github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: finding gopkg.in/olivere/elastic.v5 v5.0.80
go: finding github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: finding github.com/hashicorp/vault v1.1.0
go: finding github.com/hashicorp/errwrap v1.0.0
go: finding github.com/hashicorp/go-cleanhttp v0.5.1
go: finding github.com/hashicorp/go-multierror v1.0.0
go: finding github.com/hashicorp/go-retryablehttp v0.5.3
go: finding github.com/hashicorp/go-rootcerts v1.0.1
go: finding github.com/hashicorp/hcl v1.0.0
go: finding github.com/hashicorp/go-sockaddr v1.0.2
go: finding github.com/ryanuber/go-glob v1.0.0
go: finding github.com/mitchellh/mapstructure v1.1.2
go: finding github.com/minio/sio v0.2.0
go: finding github.com/coredns/coredns v1.4.0
go: finding github.com/fatih/structs v1.1.0
go: finding gopkg.in/yaml.v2 v2.2.2
go: finding github.com/klauspost/pgzip v1.2.1
go: finding github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: finding github.com/alecthomas/participle v0.2.1
go: finding github.com/minio/parquet-go v0.0.0-20190318185229-9d767baf1679
go: finding git.apache.org/thrift.git v0.12.0
go: finding github.com/pkg/profile v1.3.0
go: finding github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
go: finding github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
go: finding github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: finding github.com/prometheus/common v0.2.0
go: finding github.com/matttproud/golang_protobuf_extensions v1.0.1
go: finding github.com/prometheus/procfs v0.0.0-20190117184657-bf6a532e95b1
go: finding github.com/rs/cors v1.6.0
go: finding go.uber.org/atomic v1.3.2
# github.com/minio/minio/cmd
cmd/server-main.go:368:38: globalEndpoints.First undefined (type EndpointZones has no field or method First)
make: *** [build] Error 2

```

--- a/cmd/bootstrap-peer-server.go
+++ b/cmd/bootstrap-peer-server.go
@@ -1,0 +1,257 @@
+/*
+ * MinIO Cloud Storage, (C) 2019 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"runtime"
+	"sync/atomic"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/minio/minio-go/pkg/set"
+	xhttp "github.com/minio/minio/cmd/http"
+	"github.com/minio/minio/cmd/logger"
+	"github.com/minio/minio/cmd/rest"
+)
+
+const (
+	bootstrapRESTVersion       = "v1"
+	bootstrapRESTVersionPrefix = SlashSeparator + bootstrapRESTVersion
+	bootstrapRESTPrefix        = minioReservedBucketPath + "/bootstrap"
+	bootstrapRESTPath          = bootstrapRESTPrefix + bootstrapRESTVersionPrefix
+)
+
+const (
+	bootstrapRESTMethodVerify = "/verify"
+)
+
+// To abstract a node over network.
+type bootstrapRESTServer struct{}
+
+// ServerSystemConfig - captures information about server configuration.
+type ServerSystemConfig struct {
+	MinioPlatform  string
+	MinioRuntime   string
+	MinioEndpoints EndpointZones
+}
+
+// Diff - returns error on first difference found in two configs.
+func (s1 ServerSystemConfig) Diff(s2 ServerSystemConfig) error {
+	if s1.MinioPlatform != s2.MinioPlatform {
+		return fmt.Errorf("Expected platform '%s', found to be running '%s'",
+			s1.MinioPlatform, s2.MinioPlatform)
+	}
+	if s1.MinioEndpoints.Nodes() != s2.MinioEndpoints.Nodes() {
+		return fmt.Errorf("Expected number of endpoints %d, seen %d", s1.MinioEndpoints.Nodes(),
+			s2.MinioEndpoints.Nodes())
+	}
+
+	for i, ep := range s1.MinioEndpoints {
+		if ep.SetCount != s2.MinioEndpoints[i].SetCount {
+			return fmt.Errorf("Expected set count %d, seen %d", ep.SetCount,
+				s2.MinioEndpoints[i].SetCount)
+		}
+		if ep.DrivesPerSet != s2.MinioEndpoints[i].DrivesPerSet {
+			return fmt.Errorf("Expected drives pet set %d, seen %d", ep.DrivesPerSet,
+				s2.MinioEndpoints[i].DrivesPerSet)
+		}
+		for j, endpoint := range ep.Endpoints {
+			if endpoint.String() != s2.MinioEndpoints[i].Endpoints[j].String() {
+				return fmt.Errorf("Expected endpoint %s, seen %s", endpoint,
+					s2.MinioEndpoints[i].Endpoints[j])
+			}
+		}
+
+	}
+	return nil
+}
+
+func getServerSystemCfg() ServerSystemConfig {
+	return ServerSystemConfig{
+		MinioPlatform:  fmt.Sprintf("OS: %s | Arch: %s", runtime.GOOS, runtime.GOARCH),
+		MinioEndpoints: globalEndpoints,
+	}
+}
+
+func (b *bootstrapRESTServer) VerifyHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "VerifyHandler")
+	cfg := getServerSystemCfg()
+	logger.LogIf(ctx, json.NewEncoder(w).Encode(&cfg))
+	w.(http.Flusher).Flush()
+}
+
+// registerBootstrapRESTHandlers - register bootstrap rest router.
+func registerBootstrapRESTHandlers(router *mux.Router) {
+	server := &bootstrapRESTServer{}
+	subrouter := router.PathPrefix(bootstrapRESTPrefix).Subrouter()
+
+	subrouter.Methods(http.MethodPost).Path(bootstrapRESTVersionPrefix + bootstrapRESTMethodVerify).HandlerFunc(
+		httpTraceHdrs(server.VerifyHandler))
+}
+
+// client to talk to bootstrap Nodes.
+type bootstrapRESTClient struct {
+	endpoint   Endpoint
+	restClient *rest.Client
+	connected  int32
+}
+
+// Reconnect to a bootstrap rest server.k
+func (client *bootstrapRESTClient) reConnect() {
+	atomic.StoreInt32(&client.connected, 1)
+}
+
+// Wrapper to restClient.Call to handle network errors, in case of network error the connection is marked disconnected
+// permanently. The only way to restore the connection is at the xl-sets layer by xlsets.monitorAndConnectEndpoints()
+// after verifying format.json
+func (client *bootstrapRESTClient) call(method string, values url.Values, body io.Reader, length int64) (respBody io.ReadCloser, err error) {
+	return client.callWithContext(context.Background(), method, values, body, length)
+}
+
+// Wrapper to restClient.Call to handle network errors, in case of network error the connection is marked disconnected
+// permanently. The only way to restore the connection is at the xl-sets layer by xlsets.monitorAndConnectEndpoints()
+// after verifying format.json
+func (client *bootstrapRESTClient) callWithContext(ctx context.Context, method string, values url.Values, body io.Reader, length int64) (respBody io.ReadCloser, err error) {
+	if !client.IsOnline() {
+		client.reConnect()
+	}
+
+	if values == nil {
+		values = make(url.Values)
+	}
+
+	respBody, err = client.restClient.CallWithContext(ctx, method, values, body, length)
+	if err == nil {
+		return respBody, nil
+	}
+
+	if isNetworkError(err) {
+		atomic.StoreInt32(&client.connected, 0)
+	}
+
+	return nil, err
+}
+
+// Stringer provides a canonicalized representation of node.
+func (client *bootstrapRESTClient) String() string {
+	return client.endpoint.String()
+}
+
+// IsOnline - returns whether RPC client failed to connect or not.
+func (client *bootstrapRESTClient) IsOnline() bool {
+	return atomic.LoadInt32(&client.connected) == 1
+}
+
+// Close - marks the client as closed.
+func (client *bootstrapRESTClient) Close() error {
+	atomic.StoreInt32(&client.connected, 0)
+	client.restClient.Close()
+	return nil
+}
+
+// Verify - fetches system server config.
+func (client *bootstrapRESTClient) Verify(srcCfg ServerSystemConfig) (err error) {
+	if newObjectLayerFn() != nil {
+		return nil
+	}
+	respBody, err := client.call(bootstrapRESTMethodVerify, nil, nil, -1)
+	if err != nil {
+		return
+	}
+	defer xhttp.DrainBody(respBody)
+	recvCfg := ServerSystemConfig{}
+	if err = json.NewDecoder(respBody).Decode(&recvCfg); err != nil {
+		return err
+	}
+	return srcCfg.Diff(recvCfg)
+}
+
+func verifyServerSystemConfig(endpointZones EndpointZones) error {
+	srcCfg := getServerSystemCfg()
+	clnts := newBootstrapRESTClients(endpointZones)
+	var onlineServers int
+	for onlineServers < len(clnts)/2 {
+		for _, clnt := range clnts {
+			if err := clnt.Verify(srcCfg); err != nil {
+				if isNetworkError(err) {
+					continue
+				}
+				return fmt.Errorf("%s as has incorrect configuration: %w", clnt.String(), err)
+			}
+			onlineServers++
+		}
+		// Sleep for a while - so that we don't go into
+		// 100% CPU when half the endpoints are offline.
+		time.Sleep(500 * time.Millisecond)
+	}
+	return nil
+}
+
+func newBootstrapRESTClients(endpointZones EndpointZones) []*bootstrapRESTClient {
+	seenHosts := set.NewStringSet()
+	var clnts []*bootstrapRESTClient
+	for _, ep := range endpointZones {
+		for _, endpoint := range ep.Endpoints {
+			if seenHosts.Contains(endpoint.Host) {
+				continue
+			}
+			seenHosts.Add(endpoint.Host)
+
+			// Only proceed for remote endpoints.
+			if !endpoint.IsLocal {
+				clnt, err := newBootstrapRESTClient(endpoint)
+				if err != nil {
+					continue
+				}
+				clnts = append(clnts, clnt)
+			}
+		}
+	}
+	return clnts
+}
+
+// Returns a new bootstrap client.
+func newBootstrapRESTClient(endpoint Endpoint) (*bootstrapRESTClient, error) {
+	serverURL := &url.URL{
+		Scheme: endpoint.Scheme,
+		Host:   endpoint.Host,
+		Path:   bootstrapRESTPath,
+	}
+
+	var tlsConfig *tls.Config
+	if globalIsSSL {
+		tlsConfig = &tls.Config{
+			ServerName: endpoint.Hostname(),
+			RootCAs:    globalRootCAs,
+		}
+	}
+
+	trFn := newCustomHTTPTransport(tlsConfig, rest.DefaultRESTTimeout, rest.DefaultRESTTimeout)
+	restClient, err := rest.NewClient(serverURL, trFn, newAuthToken)
+	if err != nil {
+		return nil, err
+	}
+
+	return &bootstrapRESTClient{endpoint: endpoint, restClient: restClient, connected: 1}, nil
+}

--- a/cmd/endpoint.go
+++ b/cmd/endpoint.go
@@ -196,8 +196,8 @@ type ZoneEndpoints struct {
 // EndpointZones - list of list of endpoints
 type EndpointZones []ZoneEndpoints
 
-// First returns true if the first endpoint is local.
-func (l EndpointZones) First() bool {
+// FirstLocal returns true if the first endpoint is local.
+func (l EndpointZones) FirstLocal() bool {
 	return l[0].Endpoints[0].IsLocal
 }
 

--- a/cmd/lock-rest-server.go
+++ b/cmd/lock-rest-server.go
@@ -149,8 +149,4 @@ func registerLockRESTHandlers(router *mux.Router, endpointZones EndpointZones) {
 			globalLockServers[endpoint] = lockServer.ll
 		}
 	}
-
-	// If none of the routes match add default error handler routes
-	router.NotFoundHandler = http.HandlerFunc(httpTraceAll(errorResponseHandler))
-	router.MethodNotAllowedHandler = http.HandlerFunc(httpTraceAll(errorResponseHandler))
 }

--- a/cmd/peer-rest-server.go
+++ b/cmd/peer-rest-server.go
@@ -1086,11 +1086,6 @@ func registerPeerRESTHandlers(router *mux.Router) {
 	subrouter.Methods(http.MethodPost).Path(peerRESTVersionPrefix + peerRESTMethodTrace).HandlerFunc(server.TraceHandler)
 	subrouter.Methods(http.MethodPost).Path(peerRESTVersionPrefix + peerRESTMethodBackgroundHealStatus).HandlerFunc(server.BackgroundHealStatusHandler)
 	subrouter.Methods(http.MethodPost).Path(peerRESTVersionPrefix + peerRESTMethodLog).HandlerFunc(server.ConsoleLogHandler)
-
 	subrouter.Methods(http.MethodPost).Path(peerRESTVersionPrefix + peerRESTMethodPutBucketObjectLockConfig).HandlerFunc(httpTraceHdrs(server.PutBucketObjectLockConfigHandler)).Queries(restQueries(peerRESTBucket)...)
 	subrouter.Methods(http.MethodPost).Path(peerRESTVersionPrefix + peerRESTMethodBucketObjectLockConfigRemove).HandlerFunc(httpTraceHdrs(server.RemoveBucketObjectLockConfigHandler)).Queries(restQueries(peerRESTBucket)...)
-
-	// If none of the routes match add default error handler routes
-	router.NotFoundHandler = http.HandlerFunc(httpTraceAll(errorResponseHandler))
-	router.MethodNotAllowedHandler = http.HandlerFunc(httpTraceAll(errorResponseHandler))
 }

--- a/cmd/routers.go
+++ b/cmd/routers.go
@@ -24,15 +24,17 @@ import (
 
 // Composed function registering routers for only distributed XL setup.
 func registerDistXLRouters(router *mux.Router, endpointZones EndpointZones) {
-	// Register storage rpc router only if its a distributed setup.
+	// Register storage REST router only if its a distributed setup.
 	registerStorageRESTHandlers(router, endpointZones)
 
 	// Register peer REST router only if its a distributed setup.
 	registerPeerRESTHandlers(router)
 
-	// Register distributed namespace lock.
-	registerLockRESTHandlers(router, endpointZones)
+	// Register bootstrap REST router for distributed setups.
+	registerBootstrapRESTHandlers(router)
 
+	// Register distributed namespace lock routers.
+	registerLockRESTHandlers(router, endpointZones)
 }
 
 // List of some generic handlers which are applied for all incoming requests.
@@ -111,6 +113,10 @@ func configureServerHandler(endpointZones EndpointZones) (http.Handler, error) {
 	// Add API router, additionally all server mode support encryption
 	// but don't allow SSE-KMS.
 	registerAPIRouter(router, true, false)
+
+	// If none of the routes match add default error handler routes
+	router.NotFoundHandler = http.HandlerFunc(httpTraceAll(errorResponseHandler))
+	router.MethodNotAllowedHandler = http.HandlerFunc(httpTraceAll(errorResponseHandler))
 
 	// Register rest of the handlers.
 	return registerHandlers(router, globalHandlers...), nil

--- a/cmd/server-main.go
+++ b/cmd/server-main.go
@@ -365,6 +365,13 @@ func serverMain(ctx *cli.Context) {
 		globalHTTPServerErrorCh <- globalHTTPServer.Start()
 	}()
 
+	if globalIsDistXL && globalEndpoints.FirstLocal() {
+		// Additionally in distributed setup validate
+		if err := verifyServerSystemConfig(globalEndpoints); err != nil {
+			logger.Fatal(err, "Unable to initialize distributed setup")
+		}
+	}
+
 	newObject, err := newObjectLayer(globalEndpoints)
 	logger.SetDeploymentID(globalDeploymentID)
 	if err != nil {

--- a/cmd/storage-rest-server.go
+++ b/cmd/storage-rest-server.go
@@ -612,8 +612,4 @@ func registerStorageRESTHandlers(router *mux.Router, endpointZones EndpointZones
 				Queries(restQueries(storageRESTVolume, storageRESTFilePath, storageRESTBitrotAlgo, storageRESTBitrotHash, storageRESTLength, storageRESTShardSize)...)
 		}
 	}
-
-	// If none of the routes match add default error handler routes
-	router.NotFoundHandler = http.HandlerFunc(httpTraceAll(errorResponseHandler))
-	router.MethodNotAllowedHandler = http.HandlerFunc(httpTraceAll(errorResponseHandler))
 }

--- a/cmd/xl-zones.go
+++ b/cmd/xl-zones.go
@@ -60,7 +60,7 @@ func newXLZones(endpointZones EndpointZones) (ObjectLayer, error) {
 		z       = &xlZones{zones: make([]*xlSets, len(endpointZones))}
 	)
 	for i, ep := range endpointZones {
-		formats[i], err = waitForFormatXL(endpointZones.First(), ep.Endpoints,
+		formats[i], err = waitForFormatXL(endpointZones.FirstLocal(), ep.Endpoints,
 			ep.SetCount, ep.DrivesPerSet, deploymentID)
 		if err != nil {
 			return nil, err

